src/main/java/org/opentripplanner/routing/algorithm/filterchain/GroupBySimilarity.java

```diff
@@ -43,23 +43,21 @@
   public final double groupByP;
 
   /**
-   * Set a guideline for the minimum total number of itineraries to keep. For example, if there is 2
-   * groups and the min-limit is 3, the filter will keep 2 samples in each group, keeping up to 4
-   * itineraries(approximately 3). Set this to 1 to keep ONE itinerary per group.
+   * Set a maximum number of itineraries to keep for each group.
    */
-  public final int approximateMinLimit;
+  public final int maxNumOfItinerariesPrGroup;
 
 
-  public GroupBySimilarity(double groupByP, int approximateMinLimit) {
+  public GroupBySimilarity(double groupByP, int maxNumOfItinerariesPrGroup) {
     this.groupByP = groupByP;
-    this.approximateMinLimit = approximateMinLimit;
+    this.maxNumOfItinerariesPrGroup = maxNumOfItinerariesPrGroup;
   }
 
   @Override
   public String toString() {
     return ToStringBuilder.of(GroupBySimilarity.class)
         .addNum("groupByP", groupByP)
-        .addNum("approximateMinLimit", approximateMinLimit)
+        .addNum("approximateMinLimit", maxNumOfItinerariesPrGroup)
```

Comment:
```suggestion
        .addNum("maxNumOfItinerariesPerGroup", maxNumOfItinerariesPrGroup)
````

Reply:
I also changed the name to maxNumOfItinerariesPerGroup (Pr -> Per)